google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClientImpl.java

```diff
@@ -39,11 +42,32 @@
     this.pool = pool;
   }
 
+  @VisibleForTesting
+  PooledSession getReadSession() {
+    return pool.getReadSession();
+  }
+
+  @VisibleForTesting
+  PooledSession getReadWriteSession() {
+    return pool.getReadWriteSession();
+  }
+
+  @VisibleForTesting
+  int getNumberOfAvailableWritePreparedSessions() {
```

Comment:
this is not DBClient's responsibility. I wouldn't expose it here

Reply:
Method is now removed and the visibility of the session pool changed to package-private to make it visible for test cases.